homeassistant/components/hue_ble/light.py

```diff
@@ -0,0 +1,183 @@
+"""light platform."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+from HueBLE import HueBleLight
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP_KELVIN,
+    ATTR_XY_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.util import color as color_util
+
+_LOGGER = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from . import HueBLEConfigEntry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HueBLEConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Add light for passed config_entry in HA."""
+
+    light = config_entry.runtime_data
+    async_add_entities([HaHueBLE(light)])
+
+
+class HaHueBLE(LightEntity):
+    """Representation of a light."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, api: HueBleLight) -> None:
+        """Initialize the light object. Does not connect."""
+
+        self._light = api
+        self._name = self._light.name
+        self._address = self._light.address
+        self._attr_unique_id = self._light.address
+        self._attr_available = self._light.available
+        self._attr_is_on = self._light.power_state
+        self._attr_brightness = self._light.brightness
+        self._attr_color_temp_kelvin = color_util.color_temperature_mired_to_kelvin(
+            self._light.colour_temp
+        )
+        self._attr_xy_color = self._light.colour_xy
+        self._attr_min_color_temp_kelvin = color_util.color_temperature_mired_to_kelvin(
+            self._light.maximum_mireds
+        )
+        self._attr_max_color_temp_kelvin = color_util.color_temperature_mired_to_kelvin(
+            self._light.minimum_mireds
+        )
```

Comment:
I would assume that this code is also used for normal updates. I would put that in a separate method to DRY